bake/installation-configuration.sh

```diff
@@ -78,6 +77,25 @@ function wait_for_api {
 
 wait_for_api
 
+wait_approve_csr() {
```

Comment:
What's this for?

Reply:
When `kubelet` starts and the hostname of the node has changed, it needs to register it as a new node. In order to be authorized to do so, those 2 CSRs need to be approved. 

(those certs are described [here](https://kubernetes.io/docs/reference/access-authn-authz/kubelet-tls-bootstrapping/#client-and-serving-certificates))

What led me here are the following apiserver errors, after starting the baked image with `make start-vm HOSTNAME=foobar`:

```shell
Jun 22 09:33:55 foobar kubenswrapper[2522]: E0622 09:33:55.405280    2522 kubelet_node_status.go:94] "Unable to
 register node with API server" err="nodes \"foobar\" is forbidden: node \"master1\" is not allowed to modify node
 \"foobar\"" node="foobar"

Jun 22 09:34:00 foobar kubenswrapper[2522]: I0622 09:34:00.504372    2522 csi_plugin.go:913] Failed to contact API
 server when waiting for CSINode publishing: csinodes.storage.k8s.io "foobar" is forbidden: User
 "system:node:master1" cannot get resource "csinodes" in API group "storage.k8s.io" at the cluster scope: can
 only access CSINode with the same name as the requesting node
```

Reply:
Will this be skipped if I don't change the hostname?

Reply:
No, it's not a no-op, I'm looking into it though, thanks for pointing this out :+1: 

Other than that, does this seem reasonable to you? Do you think there is anything else we could leverage or another way altogether?

Reply:
If relocation it's good but for image-based it's gonna waste too much time, we need to force change the hostname in all resources... Might be a huge undertaking

Reply:
Or maybe disallow changing the hostname altogether for image-based, but I'm not sure users would love this limitation

Reply:
>For relocation it's good but for image-based it's gonna waste too much time. Instead we need to force change the hostname in all resources... Which might be a huge undertaking but I don't see a way around that

That's my thinking as well, let's focus on relocation and I'll try to verify if force-changing the hostname everywhere  is as big an undertaking as we think it is.

 > Or maybe disallow changing the hostname altogether for image-based, but I'm not sure users would love this limitation
Not sure either, let's have an acceptable solution for it and I guess we'll find out whether we could drop this feature or not in the future :man_shrugging: 

I'll refactoring these operations to no-op when the hostname hasn't changed and we can discuss this topic on our sync as well.